pytorch_lightning/trainer/properties.py

```diff
@@ -254,6 +255,12 @@ def progress_bar_dict(self) -> dict:
         all_metrics.update(**logged_metrics)
         return all_metrics
 
+    @property
+    def should_reload_dl_epoch(self) -> bool:
```

Comment:
Do you see this as a property that users might want to rely on?

If not, I wouldn't add this logic here

Reply:
Perhaps. What do the others think?  @tchaton [mentioned](https://github.com/PyTorchLightning/pytorch-lightning/pull/5043#discussion_r539507110) in a review earlier if this logic could become a property of the trainer

Reply:
It can be in the trainer (at least for now) but let's mark it as protected so we don't have to deprecate later

`should_reload_dl_epoch` -> `_should_reload_dl_epoch`

Reply:
Sure